server/ladder_service.py

```diff
@@ -365,15 +375,75 @@ def on_match_found(
 
                 self._clear_search(player, queue.name)
 
-            asyncio.create_task(
-                self.start_game(s1.players, s2.players, queue)
-            )
+            asyncio.create_task(self.confirm_match(s1, s2, queue))
         except Exception:
             self._logger.exception(
                 "Error processing match between searches %s, and %s",
                 s1, s2
             )
 
+    async def confirm_match(
+        self,
+        s1: Search,
+        s2: Search,
+        queue: MatchmakerQueue
+    ):
+        try:
+            all_players = s1.players + s2.players
+
+            offer = self.create_match_offer(all_players)
+            offer.write_broadcast_update()
+            await offer.wait_ready()
+
+            await self.start_game(s1.players, s2.players, queue)
+        except OfferTimeoutError:
+            unready_players = list(offer.get_unready_players())
+            self._logger.info(
+                "Match failed to start. Some players did not ready up in time: %s",
+                [player.login for player in unready_players]
+            )
+            # TODO: Refactor duplication
+            msg = {"command": "match_cancelled"}
+            for player in all_players:
+                if player.state == PlayerState.STARTING_AUTOMATCH:
+                    player.state = PlayerState.IDLE
+                player.write_message(msg)
+
+            # Return any player that accepted the match back to the queue
+            # TODO: make this work with parties
```

Comment:
Probably don't need to even add auto-requeue to begin with. It would be nice, but if its too hard, people can figure out how to requeue on their own.